<div class="alert" style="background: linear-gradient(to right, #ff7e5f, #feb47b); color: white;">1. import libraries!</div>


In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore') # this is for ignore warning message if any method is maybe in future are going to deprecated



 
<div style="background-color: #ffcccb; color: #900; padding: 10px; border-radius: 5px;">
2. get data
</div>


In [2]:
df = pd.read_csv('data.csv')

In [3]:
df.head()

Email  \
0      mstephenson@fernandez.com   
1              hduke@hotmail.com   
2               pallen@yahoo.com   
3        riverarebecca@gmail.com   
4  mstephens@davidson-herman.com   

                                             Address            Avatar  \
0       835 Frank Tunnel\nWrightmouth, MI 82180-9605            Violet   
1     4547 Archer Common\nDiazchester, CA 06566-8576         DarkGreen   
2  24645 Valerie Unions Suite 582\nCobbborough, D...            Bisque   
3   1414 David Throughway\nPort Jason, OH 22070-1220       SaddleBrown   
4  14023 Rodriguez Passage\nPort Jacobville, PR 3...  MediumAquaMarine   

   Avg. Session Length  Time on App  Time on Website  Length of Membership  \
0            34.497268    12.655651        39.577668              4.082621   
1            31.926272    11.109461        37.268959              2.664034   
2            33.000915    11.330278        37.110597              4.104543   
3            34.305557    13.717514        36.721283              3.120179   
4            33.330673    12.795189        37.536653              4.446308   

   Yearly Amount Spent  
0           587.951054  
1           392.204933  
2           487.547505  
3           581.852344  
4           599.406092

<div class="alert" style="
    background: linear-gradient(135deg, #ff416c 0%, #ff4b2b 100%);
    color: #fff;
    font-weight: bold;
    padding: 12px;
    border-radius: 10px;
">
  🔥 Details about columns!


1. Email = email of user.

2. Address - address of user .             

3. Avatar   - profile of user .

4. Avg. Session Length   - how long user staying on platfom [IN A MINUTES].

5. Time on App- how long user spend time on the app [IN A MINUTES].

6. Time on Website - how long user spend time on the website [IN A MINUTES].

7. Length of Membership-from how long user is member of that platform [IN A YEAR].

8. Yearly Amount Spent - how much total AMOUNTS user spend on the platform [IN A $].


</div>



<div class="alert" style="
    background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
    color: #fff;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    border-radius: 8px;
">
  3. do basic operation to understand data
</div>


In [4]:
df.shape

(500, 8)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Email                 500 non-null    object 
 1   Address               500 non-null    object 
 2   Avatar                500 non-null    object 
 3   Avg. Session Length   500 non-null    float64
 4   Time on App           500 non-null    float64
 5   Time on Website       500 non-null    float64
 6   Length of Membership  500 non-null    float64
 7   Yearly Amount Spent   500 non-null    float64
dtypes: float64(5), object(3)
memory usage: 31.4+ KB


In [10]:
df.duplicated().sum()

np.int64(0)

<div class="alert" style="
    background-color: #ffe0b2;
    color: #ff6f00;
    border: 1px solid #ff6f00;
    padding: 12px;
    border-radius: 6px;
">
  ⚠️ Remove unwanted columns which dont help ML.
</div>


In [11]:
len(df['Email'].unique())

500

In [12]:
len(df['Address'].unique())

500

In [13]:
len(df['Avatar'].unique())

138

In [14]:
len(df['Avg. Session Length'].unique())

500

In [15]:
len(df['Time on App'].unique())

500

In [16]:
len(df['Time on Website'].unique())


500

In [17]:
len(df['Length of Membership'].unique())

500

In [18]:
len(df['Yearly Amount Spent'].unique())

500

<div class="alert" style="
    background: linear-gradient(135deg, #f4ff81 0%, #c6ff00 100%);
    color: #333;
    box-shadow: 0 0 12px rgba(198, 255, 0, 0.7);
    padding: 12px;
    border-radius: 8px;
">
  🌟 here objects columns remove because its can't help our model !, Reason?

1. Email are all unique.

2. Address are also all unique if its Availabel in 
   region wise so its help but here is not a case.

3. Avatar - is 138 but my data set is so small so , 
   even i do encoding its doesn't make any sense and also 138 categories preety high (if its avilabe <10 so we can do ) , and even alot of user doesn't set avatar by self it set by default by app/website.
   
</div>
